plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1400 @@
+        target_list = []
+        target = config.get('host')
+        have_targetentry = have.get('host')
+        have_targetentry_names = []
+        if have_targetentry and not overridden_or_replaced:
+            have_targetentry_names = [have_entry.get('name') for have_entry in have_targetentry]
+
+        for conf in target:
+            target_dict = {}
+            retries = conf.get('retries')
+            if retries is None:
+                retries = 3
+            target_entry_name = ""
+            if 'name' in conf:
+                target_entry_name = conf.get('name')
+            else:
+                target_entry_name = self.get_targetentry(have_targetentry_names)
+            target_dict['name'] = target_entry_name
+            target_dict['retries'] = retries
+            tag_list = []
+            tag = None
+            if 'tag' in conf:
+                tag = str(conf.get('tag')) + "Notify"
+            tag_list.append(tag)
+            if conf.get('vrf') and 'tag' in conf:
+                tag_list.append(str(conf.get('vrf')))
+            if len(tag_list) == 0:
+                tag_list = None
+
+            target_dict["tag"] = tag_list
+
+            target_dict['target-params'] = target_entry_name
+            target_dict['timeout'] = conf.get('timeout')
+            port = conf.get('port')
+            if not port:
+                port = 162
+            target_dict['udp'] = {'ip': conf.get('ip'), 'port': port}
+            target_dict['ietf-snmp-ext:source-interface'] = conf.get('source_interface')
+            target_list.append(target_dict)
+
+        payload_url['target'] = target_list
+        return payload_url
+
+    def build_create_enable_target_params_payload(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP param information based on the given host configuration
+        :rtype: A dictionary
+        :returns: The payload for SNMP target
+        """
+        payload_url = {}
+        requests = []
+        target_params_list = []
+
+        server = config.get('host')
+        have_targetentry = have.get('host')
+        have_targetentry_names = []
+        if have_targetentry and not overridden_or_replaced:
+            have_targetentry_names = [agent_entry.get('name') for agent_entry in have_targetentry]
+        for conf in server:
+            target_params_dict = {}
+            target_entry_name = ""
+            matched_host = None
+            if 'name' in conf:
+                target_entry_name = conf.get('name')
+                matched_host = next((x for x in have_targetentry if x.get('name') == target_entry_name), None)
+            elif have_targetentry:
+                matched_host = next((host for host in have_targetentry if host.get('ip') == conf.get('ip')
+                                     and host.get('vrf') == conf.get('vrf')), None)
+                if matched_host:
+                    target_entry_name = matched_host['name']
```

Comment:
Save this result so it can be used by build_create_enable_target_payload.

```suggestion
                    target_entry_name = matched_host['name']
                    conf['name'] = target_entry_name
```